gcapi/client.py

```diff
@@ -715,103 +728,116 @@ def run_external_job(  # noqa: C901
+        items = list(
+            client.reader_studies.display_sets.iterate_all(
+                params={"reader_study": reader_study.pk}
+            )
+        )
+
+        To then add, for example, a PDF report and a lung volume
+        value to the first archive item , provide the interface slugs together
+        with the respective value or file path as follows:
+        client.update_display_set(
+            display_set_pk=items[0].id,
+            values={
+                "report": [...],
+                "lung-volume": 1.9,
             },
-            ...
-        ]
+        )
+        If you provide a value or file for an existing interface of the display
+        set, the old value will be overwritten by the new one, hence allowing you
+        to update existing display-set values.
 
-        Where the file paths are local paths to the files making up a
-        single image. For file type interface the file path can only
-        contain a single file. For json type interfaces any value that
-        is valid for the interface can be passed.`
+        Parameters
+        display_set_pk
+        values
+
+        Returns
+        -------
+        The updated display set
+        """
+        ds = yield from self.__org_api_meta.reader_studies.display_sets.detail(
+            pk=display_set_pk
+        )
+        return (
+            yield from self._update_socket_value_set(
+                target=ds,
+                description=values,
+                api=self.__org_api_meta.reader_studies.display_sets,
+            )
+        )
+
+    def add_cases_to_reader_study(
+        self,
+        *,
+        reader_study: Union[str, gcapi.models.ReaderStudy],
+        display_sets: list[SocketValueSetDescription],
+    ):
+        """
+        This function takes an reader-study slug or model and a list of display-set
+        descriptions. It then creates the display-sets for the reader study.
 
         Parameters
         ----------
         reader_study
+            slug for the reader study (e.g. "i-am-a-reader-study"). You can find this
+            readily in the URL you use to visit the archive page:
+
+                https://grand-challenge.org/reader-studies/i-am-a-reader-study/
+
         display_sets
+            The format for the description of display sets is as follows:
+
+            [
+                {
+                    "slug_0": ["filepath_0", ...],
+                    "slug_1": "filepath_0",
+                    "slug_2": pathlib.Path("filepath_0"),
+                    ...
+                    "slug_n": {"json": "value"}
+
+                },
+                ...
+            ]
+
+            Where the file paths are local paths to the files making up a
+            single image. For file-kind sockets the file path can only
+            reference a single file. For json-kind sockets any value that
+            is valid for the sockets can directly be passed.
```

Comment:
Again maybe worth adding an example of how one can pass in an existing CIV?

Reply:
Same above for the other helper funcion

Reply:
That is a good suggestion. I've added a quick snippet on images and also made a notion of value-kinds kept in files.